### PR TITLE
test: run linux package tests nightly against latest contrib main

### LINKED ISSUE #659: Add nightly linux package tests
### Description
Create a nightly version of the linux package tests that use the latest snapshots of main branch of core and contrib distributions from the respective repos.

This is a follow-up to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34372

### PR DESCRIPTION
**Pull request by mowies**
This PR enables nightly testing of the otelcol-contrib linux packages based on the latest main branch version of otelcol-contrib built in https://github.com/open-telemetry/opentelemetry-collector-contrib.
This change was made possible by splitting the otelcol-contrib gorelease config into 2 separate files, one for building and one for packaging/archives/docker images/etc.
(The split goreleaser build is a goreleaser pro feature)

Using the split building approach was the only viable option to achieve this. I did lots of research on this, it was talked about in the Collector SIG meeting and I also [asked the GoReleaser folks for their help](https://github.com/goreleaser/goreleaser/issues/5234).

Consequences of this change:
- better testing
- slightly less maintainability of the goreleaser files and generator code for otelcol-contrib since they have special handling now

Fixes #659

### CODE REVIEW

File: scripts/build.sh
Comment (mowies) [inline scripts/build.sh:9]: removed unused flag

File: .github/workflows/base-package-tests.yaml
Comment (mx-psi) [inline .github/workflows/base-package-tests.yaml:16]: Should we split this into a separate PR?

File: .github/workflows/base-ci-goreleaser.yaml
Comment (mx-psi) [inline .github/workflows/base-ci-goreleaser.yaml:57]: Is this something we can add independently from this PR?

File: .github/workflows/base-package-tests.yaml
Comment (mowies) [reply .github/workflows/base-package-tests.yaml]: removed

File: .github/workflows/base-ci-goreleaser.yaml
Comment (mowies) [reply .github/workflows/base-ci-goreleaser.yaml]: reverted

File: .github/workflows/base-ci-goreleaser.yaml
Comment (mowies) [reply .github/workflows/base-ci-goreleaser.yaml]: reverted

File: .github/workflows/base-ci-goreleaser.yaml
Comment (mx-psi) [inline .github/workflows/base-ci-goreleaser.yaml:86]: What happens if a run id is not found?

File: .github/workflows/package-test.yaml
Comment (mx-psi) [inline .github/workflows/package-test.yaml:1]: How will we be able to tell if there is an error? I think we should create a `release:blocker` issue if there is a failed test. We can do this in a follow up PR if you want to.

File: .github/workflows/base-ci-goreleaser.yaml
Comment (mowies) [reply .github/workflows/base-ci-goreleaser.yaml:86]: There will always be a `build-and-test` run that succeeded. I double checked things like pagination and limits set up in the GH CLI and also in the GH API behind and it should always be fine.

File: .github/workflows/package-test.yaml
Comment (mowies) [reply .github/workflows/package-test.yaml:1]: Yes, that's a good and valid point. I think I'll do a follow up for that, just to be sure that everything runs as expected after merging and we don't get spammy issues.

Comment (mowies) [reply .github/workflows/package-test.yaml:1]: Would the issue be created in the -contrib repo or in -releases?

File: cmd/goreleaser/internal/configure.go
Comment (mowies) [reply cmd/goreleaser/internal/configure.go:139]: I just pulled this out of the parent function, it was there before. See https://github.com/open-telemetry/opentelemetry-collector-releases/pull/666/files#diff-d211e0dff55b99aa85288ba7b78152b46dda3763a5f8f5935848b6a94ff6ae9cL92

File: cmd/goreleaser/main.go
Comment (mowies) [reply cmd/goreleaser/main.go:29]: naming is hard 😅 i like `--generate-build-step`

File: .github/workflows/base-ci-goreleaser.yaml
Comment (mowies) [reply .github/workflows/base-ci-goreleaser.yaml]: done

File: .github/workflows/base-ci-goreleaser.yaml
Comment (mowies) [reply .github/workflows/base-ci-goreleaser.yaml:91]: I make use of this folder in both cases, nightly and normal. So it is always needed

File: .github/workflows/base-ci-goreleaser.yaml
Comment (mowies) [reply .github/workflows/base-ci-goreleaser.yaml:1]: i'll try but it's somewhat tricky with files being downloaded and used in later steps and things

File: .github/workflows/package-test.yaml
Comment (mx-psi) [reply .github/workflows/package-test.yaml:1]: Created #756 as a follow up to this PR

**Comment by mx-psi** [comment]
I have re-run the CI but it keeps failing, can you take a look? Do you think it is related to the 2.4.6 update?

**Comment by mowies** [comment]
yes i think it's related somehow. there were some changes to folder naming that i need to adapt to I believe. This will also be needed on the renovate PR

**Comment by mowies** [comment]
Maybe i will wait until we merge the updated goreleaser version in main. That will require some manual changes as well I think. Maybe you could retrigger CI there and then let's see what happens.
PR link: https://github.com/open-telemetry/opentelemetry-collector-releases/pull/752

File: .github/workflows/base-ci-goreleaser.yaml
Comment (mowies) [reply .github/workflows/base-ci-goreleaser.yaml:1]: i can't find a good way to abstract this away right now. the pipeline is so interleaved that it's hard to extract steps in the middle without basically duplicating the whole thing once for otelcol-contrib and once for the rest. but i will add a "nightly" prefix to the relevant steps

**Comment by mowies** [comment]
@mx-psi i believe i'm done with all the review comments and refactoring etc.
would appreciate a re-review when you have time :)

**Comment by mowies** [comment]
you mean merging this only after the next release?
doing a release on my fork sounds good

**Comment by mx-psi** [comment]
> you mean merging this only after the next release?

Yep

> doing a release on my fork sounds good

Alright, if you find the time, I suppose it will fail when pushing to the Docker registry but if it passes until then that's a good test for me

**Comment by mowies** [comment]
Sounds good, I'll do the test release next week

**Comment by mowies** [comment]
i tried to do a test release but it's hard to do with so many otel specific things. Even after removing the container registry login steps, goreleaser needs the pro key for example. But here are the test runs that ran until the requirement for the goreleaser-pro key:

[Builder](https://github.com/mowies/opentelemetry-collector-releases/actions/runs/12115809325)
[Core](https://github.com/mowies/opentelemetry-collector-releases/actions/runs/12115809307)
[Contrib](https://github.com/mowies/opentelemetry-collector-releases/actions/runs/12115809322)
[k8s](https://github.com/mowies/opentelemetry-collector-releases/actions/runs/12115809336)
[OTLP](https://github.com/mowies/opentelemetry-collector-releases/actions/runs/12115809313)

In any case, this PR doesn't touch any of the release pipelines anyways.

**Review by atoulme — APPROVED** [review]

**Comment by mowies** [comment]
@mx-psi so now it was still merged before the collector release 😅

**Comment by mx-psi** [comment]
> @mx-psi so now it was still merged before the collector release 😅

Sorry, should have said something here before merging: I spoke with Antoine (this week's release manager) about this and he was fine to merge this before the release :smile:

**Comment by mowies** [comment]
ah okok 😄

**Pull request merged**
